carmin-server/server/resources/helpers/executions.py

```diff
@@ -1,10 +1,138 @@
+    return execution_dir_absolute_path, error
+
+
+def get_execution_dir(username: str, execution_identifier: str) -> str:
+    return os.path.join(
+        get_user_data_directory(username), EXECUTIONS_DIRNAME,
+        execution_identifier)
+
+
+def write_inputs_to_file(execution: Execution,
+                         path_to_execution_dir: str) -> ErrorCodeAndMessage:
+    inputs_json_file = os.path.join(path_to_execution_dir, INPUTS_FILENAME)
     write_content = json.dumps(execution.input_values)
-    with open(inputs_json_file, 'w') as f:
-        f.write(write_content)
+    try:
+        with open(inputs_json_file, 'w') as f:
+            f.write(write_content)
+    except OSError:
+        return UNEXPECTED_ERROR
+
+
+def input_files_exist(input_values: Dict, pipeline: Pipeline,
+                      url_root: str) -> (bool, str):
+    pipeline_parameters = pipeline.parameters
+
+    for key in input_values:
+        for parameter in pipeline_parameters:
+            if parameter.parameter_type == "File" and parameter.name == key:
+                exists, path = platform_path_exists(url_root,
+                                                    input_values[key])
+                if not exists:
+                    return False, path
+    return True, None
+
+
+def get_pipeline(pipeline_identifier: str) -> Pipeline:
+    pipeline_directory = app.config['PIPELINE_DIRECTORY']
+    all_pipelines = [
+        f for f in os.listdir(pipeline_directory) if not f.startswith(".")
+    ]
+    for pipeline in all_pipelines:
+        path = os.path.join(pipeline_directory, pipeline)
+        with open(path) as f:
+            pipeline = json.load(f)
+            if pipeline["identifier"] == pipeline_identifier:
+                return PipelineSchema().load(pipeline).data
+
+
+def load_inputs(username: str,
+                execution_identifier: str) -> (Dict, ErrorCodeAndMessage):
+    execution_inputs_absolute_path = os.path.join(
+        get_user_data_directory(username), EXECUTIONS_DIRNAME,
+        execution_identifier, INPUTS_FILENAME)
+
+    if not os.path.exists(execution_inputs_absolute_path):
+        return None, INVALID_PATH
+
+    with open(execution_inputs_absolute_path) as inputs_file:
+        inputs = json.load(inputs_file)
+        return inputs, None
+
+
+def inputs_file_path(username: str, execution_identifier: str) -> str:
+    return os.path.join(
+        get_user_data_directory(username), EXECUTIONS_DIRNAME,
+        execution_identifier, INPUTS_FILENAME)
+
+
+def get_execution_as_model(username: str, identifier: str,
+                           db_session) -> (Execution, ErrorCodeAndMessage):
+    execution_db = db_session.query(ExecutionDB).filter_by(
+        identifier=identifier).first()
+
+    if not execution_db:
+        return None, EXECUTION_NOT_FOUND
+
+    inputs, errors = load_inputs(username, execution_db.identifier)
+    if errors:
+        return None, errors
+
+    dummy_exec = Execution(None, None, None)
```

Comment:
I am not very happy of how I got to do this, but I looked for ways to access our model properties names from the class and the marshmallow schema and couldn't find any viable way.